pytransform3d/trajectories.py

```diff
@@ -436,6 +509,201 @@ def pqs_from_dual_quaternions(dqs):
     return out
 
 
+
+def norm_axis_angles(a):
+    """Normalize axis-angle representation.
+
+    Parameters
+    ----------
+    a : array-like, shape (...,4)
+        Axis of rotation and rotation angle: (x, y, z, angle)
+
+    Returns
+    -------
+    a : array, shape (...,4)
+        Axis of rotation and rotation angle: (x, y, z, angle). The length
+        of the axis vector is 1 and the angle is in [0, pi). No rotation
+        is represented by [1, 0, 0, 0].
+    """
+    angle = a[:, 3]  # Extract the angle component (N,)
+    norm = np.linalg.norm(a[:, :3], axis=1)  # Compute the norm of the first three elements (axis)
+
+
+    res = np.ones_like(a)
+
+    # Create masks for elements where angle or norm is zero
+    zero_mask = (angle == 0.0) | (norm == 0.0)
+
+
+    non_zero_mask = ~zero_mask
+    res[non_zero_mask, :3] = a[non_zero_mask, :3] / norm[non_zero_mask, None]  
```

Comment:
A more "numpy" style:

```suggestion
    res[non_zero_mask, :3] = a[non_zero_mask, :3] / norm[non_zero_mask, :]  
```